qemu/qmp.go

```diff
@@ -1419,14 +1419,22 @@ func (q *QMP) ExecMemdevAdd(ctx context.Context, qomtype, id, mempath string, si
 		"id":     driverID,
 		"memdev": id,
 	}
+
+	if bus != "" {
+		args["bus"] = bus
+	}
+	if addr != "" {
+		args["addr"] = addr
+	}
+
 	err = q.executeCommand(ctx, "device_add", args, nil)
 
 	return err
 }
 
 // ExecHotplugMemory adds size of MiB memory to the guest
-func (q *QMP) ExecHotplugMemory(ctx context.Context, qomtype, id, mempath string, size int, share bool) error {
-	return q.ExecMemdevAdd(ctx, qomtype, id, mempath, size, share, "pc-dimm", "dimm"+id)
+func (q *QMP) ExecHotplugMemory(ctx context.Context, qomtype, id, mempath string, size int, share bool, addr, bus string) error {
```

Comment:
This change probably isn't really necessary.  I don't know of any situations where regular "pc-dmm" devices aren't a special case where specifying a bus would be pointless.  It's just for virtio-mem, and other more exotic cases where we need to specify a bus, and those should go through ExecMemdevAdd instead.

Reply:
@dgibson thanks, I'll check if virtio-mem does not need it and remove the API change if so